### PR TITLE
fix(agw): c++ migration of nas/ies files

### PR DESCRIPTION
**Pull request by pruthvihebbani**
fix(agw): c++ migration of nas/ies files

## Summary

Migrated all the files under nas/ies to cpp/hpp and fixed all the compilation errors and warnings

## Test Plan

Executed s1ap integ_tests, bazel build and unit test

### CODE REVIEW

File: lte/gateway/c/core/oai/tasks/nas/ies/AdditionalUpdateType.cpp [hunk 1]
```diff
@@ -15,18 +15,23 @@
  *      contact@openairinterface.org
  */
 
+#include "lte/gateway/c/core/oai/tasks/nas/ies/AdditionalUpdateType.hpp"
 #include <stdint.h>
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/nas/ies/AdditionalUpdateType.cpp:20]: Leave the space

File: lte/gateway/c/core/oai/tasks/nas/emm/msg/NASSecurityModeCommand.hpp [hunk 2]
```diff
@@ -18,13 +18,13 @@
 #ifndef FILE_NAS_SECURITY_MODE_COMMAND_SEEN
 #define FILE_NAS_SECURITY_MODE_COMMAND_SEEN
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/nas/emm/msg/NASSecurityModeCommand.hpp]: Replace with #pragma once

File: lte/gateway/c/core/oai/tasks/nas/emm/msg/NASSecurityModeComplete.hpp [hunk 3]
```diff
@@ -18,8 +18,8 @@
 #ifndef FILE_SECURITY_MODE_COMPLETE_SEEN
 #define FILE_SECURITY_MODE_COMPLETE_SEEN
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/nas/emm/msg/NASSecurityModeComplete.hpp]: Replace with pragma once

File: lte/gateway/c/core/oai/tasks/nas/emm/msg/SecurityModeReject.hpp [hunk 4]
```diff
@@ -19,9 +19,9 @@
 #define FILE_SECURITY_MODE_REJECT_SEEN
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/nas/emm/msg/SecurityModeReject.hpp]: Replace with #pragma once

File: lte/gateway/c/core/oai/tasks/nas/emm/msg/ServiceReject.hpp [hunk 5]
```diff
@@ -20,9 +20,9 @@
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/nas/emm/msg/ServiceReject.hpp:19]: Replace with #pragma once for all headers in this PR

File: lte/gateway/c/core/oai/tasks/nas/emm/msg/TrackingAreaUpdateComplete.hpp [hunk 6]
```diff
@@ -19,8 +19,8 @@
 #define FILE_TRACKING_AREA_UPDATE_COMPLETE_SEEN
 #include <stdint.h>
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/nas/emm/msg/TrackingAreaUpdateComplete.hpp:20]: ditto

File: lte/gateway/c/core/oai/tasks/nas/emm/msg/TrackingAreaUpdateReject.hpp [hunk 7]
```diff
@@ -19,9 +19,9 @@
 #define FILE_TRACKING_AREA_UPDATE_REJECT_SEEN
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/nas/emm/msg/TrackingAreaUpdateReject.hpp]: ditto

File: lte/gateway/c/core/oai/tasks/nas/emm/msg/TrackingAreaUpdateRequest.cpp [hunk 8]
```diff
@@ -29,7 +29,7 @@ extern "C" {
 }
 #endif
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/nas/emm/msg/TrackingAreaUpdateRequest.cpp:30]: Leave space after

File: lte/gateway/c/core/oai/tasks/nas/emm/msg/UplinkNasTransport.hpp [hunk 9]
```diff
@@ -19,9 +19,9 @@
 #define FILE_UPLINK_NAS_TRANSPORT_SEEN
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/nas/emm/msg/UplinkNasTransport.hpp]: Replace with #pragma once

File: lte/gateway/c/core/oai/tasks/nas/esm/EpsBearerContextDeactivation.cpp [hunk 10]
```diff
@@ -425,7 +425,7 @@ status_code_e eps_bearer_deactivate_t3495_handler(zloop_t* loop, int timer_id,
   mme_ue_s1ap_id_t ue_id = timer_args.ue_id;
 
   ue_mm_context_t* ue_mm_context = mme_app_get_ue_context_for_timer(
-      ue_id, "EPS BEARER DEACTIVATE T3495 Timer");
+      ue_id, (char*)"EPS BEARER DEACTIVATE T3495 Timer");
```
[the remark this answers is not in the document]
Comment (rsarwad) [reply lte/gateway/c/core/oai/tasks/nas/esm/EpsBearerContextDeactivation.cpp]: Fix clang errors

File: lte/gateway/c/core/oai/tasks/nas/esm/msg/ActivateDefaultEpsBearerContextAccept.hpp [hunk 11]
```diff
@@ -23,7 +23,7 @@
 #ifdef __cplusplus
 extern "C" {
 #endif
-#include "lte/gateway/c/core/oai/tasks/nas/ies/MessageType.h"
+#include "lte/gateway/c/core/oai/tasks/nas/ies/MessageType.hpp"
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/nas/esm/msg/ActivateDefaultEpsBearerContextAccept.hpp]: Place it outside the extern C

File: lte/gateway/c/core/oai/tasks/nas/esm/msg/EsmInformationResponse.hpp [hunk 12]
```diff
@@ -23,7 +23,7 @@
 #ifdef __cplusplus
 extern "C" {
 #endif
-#include "lte/gateway/c/core/oai/tasks/nas/ies/MessageType.h"
+#include "lte/gateway/c/core/oai/tasks/nas/ies/MessageType.hpp"
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/nas/esm/msg/EsmInformationResponse.hpp]: Place it outside extern C

File: lte/gateway/c/core/oai/tasks/nas/esm/msg/ModifyEpsBearerContextAccept.hpp [hunk 13]
```diff
@@ -22,7 +22,7 @@
 #ifdef __cplusplus
 extern "C" {
 #endif
-#include "lte/gateway/c/core/oai/tasks/nas/ies/MessageType.h"
+#include "lte/gateway/c/core/oai/tasks/nas/ies/MessageType.hpp"
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/nas/esm/msg/ModifyEpsBearerContextAccept.hpp]: ditto

File: lte/gateway/c/core/oai/tasks/nas/ies/AdditionalUpdateResult.cpp [hunk 14]
```diff
@@ -15,11 +15,11 @@
  *      contact@openairinterface.org
  */
 
+#include "lte/gateway/c/core/oai/tasks/nas/ies/AdditionalUpdateResult.hpp"
 #include <stdint.h>
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/nas/ies/AdditionalUpdateResult.cpp:20]: Leave space after

File: lte/gateway/c/core/oai/tasks/nas/ies/AdditionalUpdateType.cpp [hunk 15]
```diff
@@ -15,18 +15,23 @@
  *      contact@openairinterface.org
  */
 
+#include "lte/gateway/c/core/oai/tasks/nas/ies/AdditionalUpdateType.hpp"
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/nas/ies/AdditionalUpdateType.cpp:18]: Space

File: lte/gateway/c/core/oai/tasks/nas/ies/CsfbResponse.cpp [hunk 16]
```diff
@@ -15,11 +15,11 @@
  *      contact@openairinterface.org
  */
 
+#include "lte/gateway/c/core/oai/tasks/nas/ies/CsfbResponse.hpp"
 #include <stdint.h>
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/nas/ies/CsfbResponse.cpp:20]: Space

File: lte/gateway/c/core/oai/tasks/nas/ies/DaylightSavingTime.cpp [hunk 17]
```diff
@@ -15,13 +15,13 @@
  *      contact@openairinterface.org
  */
 
+#include "lte/gateway/c/core/oai/tasks/nas/ies/DaylightSavingTime.hpp"
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/nas/ies/DaylightSavingTime.cpp:18]: Space

File: lte/gateway/c/core/oai/tasks/nas/ies/EmmCause.cpp [hunk 18]
```diff
@@ -15,11 +15,11 @@
  *      contact@openairinterface.org
  */
 
+#include "lte/gateway/c/core/oai/tasks/nas/ies/EmmCause.hpp"
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/nas/ies/EmmCause.cpp:18]: Space

File: lte/gateway/c/core/oai/tasks/nas/ies/EpsAttachResult.cpp [hunk 19]
```diff
@@ -15,11 +15,11 @@
  *      contact@openairinterface.org
  */
 
+#include "lte/gateway/c/core/oai/tasks/nas/ies/EpsAttachResult.hpp"
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/nas/ies/EpsAttachResult.cpp:18]: Space

File: lte/gateway/c/core/oai/tasks/nas/ies/NasMessageContainer.hpp [hunk 20]
```diff
@@ -20,23 +20,29 @@
...
+}
+#endif
 
 #define NAS_MESSAGE_CONTAINER_MINIMUM_LENGTH 3
 #define NAS_MESSAGE_CONTAINER_MAXIMUM_LENGTH 253
 
 typedef bstring NasMessageContainer;
 
-#ifdef __cplusplus
+/*#ifdef __cplusplus
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/nas/ies/NasMessageContainer.hpp]: Remove the commented lines

File: lte/gateway/c/core/oai/tasks/nas/ies/NasMessageContainer.hpp [hunk 21]
```diff
@@ -20,23 +20,29 @@
...
 extern "C" {
-#endif
+#endif*/
 int encode_nas_message_container(NasMessageContainer nasmessagecontainer,
                                  uint8_t iei, uint8_t* buffer, uint32_t len);
 
 int decode_nas_message_container(NasMessageContainer* nasmessagecontainer,
                                  uint8_t iei, uint8_t* buffer, uint32_t len);
-#ifdef __cplusplus
+/*#ifdef __cplusplus
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/nas/ies/NasMessageContainer.hpp]: Remove the commented lines

File: lte/gateway/c/core/oai/tasks/nas/ies/ProcedureTransactionIdentity.hpp [hunk 22]
```diff
@@ -19,6 +19,8 @@
 #define PROCEDURE_TRANSACTION_IDENTITY_H_
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/nas/ies/ProcedureTransactionIdentity.hpp]: Replace with pragma once

File: lte/gateway/c/core/oai/tasks/nas/ies/ProtocolDiscriminator.cpp [hunk 23]
```diff
@@ -15,14 +15,14 @@
  *      contact@openairinterface.org
  */
 
+#include "lte/gateway/c/core/oai/tasks/nas/ies/ProtocolDiscriminator.hpp"
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/nas/ies/ProtocolDiscriminator.cpp:18]: Space

File: lte/gateway/c/core/oai/tasks/nas/ies/TrackingAreaIdentity.cpp [hunk 24]
```diff
@@ -20,7 +20,13 @@
 
 #include "lte/gateway/c/core/oai/common/TLVEncoder.h"
 #include "lte/gateway/c/core/oai/common/TLVDecoder.h"
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/nas/ies/TrackingAreaIdentity.cpp:22]: Space

File: lte/gateway/c/core/oai/tasks/nas/ies/TrackingAreaIdentityList.cpp [hunk 25]
```diff
@@ -15,13 +15,13 @@
  *      contact@openairinterface.org
  */
 
+#include "lte/gateway/c/core/oai/tasks/nas/ies/TrackingAreaIdentityList.hpp"
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/nas/ies/TrackingAreaIdentityList.cpp:18]: Space

File: lte/gateway/c/core/oai/test/lib/test_3gpp.cpp [hunk 26]
```diff
@@ -329,4 +330,4 @@ int main(int argc, char** argv) {
   ::testing::InitGoogleTest(&argc, argv);
   OAILOG_INIT("MME", OAILOG_LEVEL_DEBUG, MAX_LOG_PROTOS);
   return RUN_ALL_TESTS();
-}
\ No newline at end of file
+}
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/test/lib/test_3gpp.cpp:333]: Check whether new line required or not

**Review by rsarwad — COMMENTED** [review]
Left some minor comments

**Comment by nstng** [comment]
Hi @pruthvihebbani, is this ready-for-review? A lot of tests are failing - should this be set as draft for now?

File: lte/gateway/c/core/oai/test/lib/test_3gpp.cpp (hunk 26 above)
Comment (pruthvihebbani) [reply lte/gateway/c/core/oai/test/lib/test_3gpp.cpp:333]: This is a formatting change only. Somehow as part of clang formatting this got removed and added

**Pull request closed without merging**